plugin/mcp-automem/scripts/process-session-memory.py

```
@@ -0,0 +1,473 @@
+            if git_branch.startswith('feature/'):
+                patterns['branch_pattern'] = 'feature_branch'
+            elif git_branch.startswith('fix/') or git_branch.startswith('bugfix/'):
+                patterns['branch_pattern'] = 'bugfix_branch'
+            elif git_branch.startswith('release/'):
+                patterns['branch_pattern'] = 'release_branch'
+            elif git_branch in ['main', 'master', 'develop']:
+                patterns['branch_pattern'] = 'main_branch'
+        
+        return patterns
+    
+    def check_duplicate(self, content: str, timeframe_hours: int = 1) -> bool:
```

Ruff (0.14.8)

Unused method argument: `timeframe_hours`

(ARG002)